src/domain/transaction/evm/evm_transaction.rs

```diff
@@ -214,6 +216,28 @@ where
         }
         Ok(())
     }
+
+    async fn calculate_total_cost(
+        &self,
```

Comment:
WDYT about modifying get_transaction_price_params to return total_cost value as part of result?

We have all the data there.